src/main/java/kubatech/api/implementations/KubaTechGTMultiBlockBase.java

```diff
@@ -0,0 +1,160 @@
+        int tiers = (int) (Math.log((double) maxInputVoltage / (double) aEUt) / ln4);
+        if (tiers <= 0) {
+            this.lEUt = aEUt;
+            this.mMaxProgresstime = aDuration;
+            return 0;
+        }
+        int durationTiers =
+                (int) (Math.log((double) aDuration / (double) getOverclockTimeLimit()) / (isPerfect ? ln4 : ln2));
+        if (durationTiers < 0) durationTiers = 0; // We do not support downclocks (yet)
+        if (durationTiers > tiers) durationTiers = tiers;
+        if (!isOverclockingInfinite()) {
+            tiers = durationTiers;
+            if (tiers == 0) {
+                this.lEUt = aEUt;
+                this.mMaxProgresstime = aDuration;
+                return 0;
+            }
+            this.lEUt = aEUt << (tiers << 1);
+            aDuration >>= isPerfect ? (tiers << 1) : tiers;
+            if (aDuration <= 0) aDuration = 1;
+            this.mMaxProgresstime = aDuration;
+            return tiers;
+        }
+        this.lEUt = aEUt << (tiers << 1);
+        aDuration >>= isPerfect ? (durationTiers << 1) : durationTiers;
+        int dMulti = tiers - durationTiers;
+        if (dMulti > 0) {
+            dMulti++;
+            // TODO: Use more inputs???
+            final ArrayList<ItemStack> stacks = new ArrayList<>(Arrays.asList(this.mOutputItems));
+            for (ItemStack mOutputItem : this.mOutputItems) {
+                mOutputItem.stackSize *= dMulti;
+                int maxSize = mOutputItem.getMaxStackSize();
+                while (mOutputItem.stackSize > maxSize)
+                    stacks.add(mOutputItem.splitStack(Math.min(mOutputItem.stackSize - maxSize, maxSize)));
+            }
+            if (stacks.size() != this.mOutputItems.length) this.mOutputItems = stacks.toArray(new ItemStack[0]);
+            for (FluidStack mOutputFluid : this.mOutputFluids) mOutputFluid.amount *= dMulti;
+        }
+        if (aDuration <= 0) aDuration = 1;
+        this.mMaxProgresstime = aDuration;
+        return tiers;
+    }
+
+    protected int calculateOverclock(long aEUt, int aDuration, boolean isPerfect) {
+        return calculateOverclock(aEUt, aDuration, getMaxInputEnergy(), isPerfect);
+    }
+
+    protected int calculateOverclock(long aEUt, int aDuration) {
+        return calculateOverclock(aEUt, aDuration, false);
+    }
+
+    protected int calculatePerfectOverclock(long aEUt, int aDuration) {
+        return calculateOverclock(aEUt, aDuration, true);
+    }
+
+    @Override
+    public long getMaxInputVoltage() {
+        return GT_ExoticEnergyInputHelper.getMaxInputVoltageMulti(getExoticAndNormalEnergyHatchList());
+    }
+
+    public long getMaxInputAmps() {
+        return GT_ExoticEnergyInputHelper.getMaxInputAmpsMulti(getExoticAndNormalEnergyHatchList());
+    }
+
+    public long getMaxInputEnergy() {
+        long energy = 0L;
+        for (GT_MetaTileEntity_Hatch hatch : getExoticAndNormalEnergyHatchList())
```

Comment:
And what is this overriden for? While the way of calculating is different it seems to not yield a different result than the base method.

Reply:
Super method is wrong

Reply:
What about it? If its wrong we should fix it

Reply:
Ah I see it. It should use average voltage

Reply:
The implementation of maxVoltage * maxAmps will return too much EU if you use multiple hatches. We should fix that